### PR TITLE
Add parsing support to point property functions

### PR DESCRIPTION
**Pull request by Ayman161803**
## Description
This PR adds parsing support to geomtry property functions. Mysql reference for the same can be found [here](https://dev.mysql.com/doc/refman/8.0/en/gis-point-property-functions.html)

## Related Issue(s)
#8604 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/sqlparser/keywords.go [hunk 1]
```diff
@@ -635,6 +635,10 @@ var keywords = []keyword{
 	{"st_polyfromwkb", ST_PolygonFromWKB},
 	{"st_polygonfromtext", ST_PolygonFromText},
 	{"st_polygonfromwkb", ST_PolygonFromWKB},
+	{"st_x", ST_X},
+	{"st_y", ST_Y},
+	{"st_latitude", ST_Latitude},
+	{"st_longitude", ST_Longitude},
```
Comment (GuptaManan100) [inline go/vt/sqlparser/keywords.go:641]: These have to stay in alphabetic order.

File: go/vt/sqlparser/sql.y [hunk 2]
```diff
@@ -807,7 +807,7 @@ query_expression:
   }
 | with_clause query_expression_parens order_by_clause limit_opt
   {
-  		$2.SetWith($1)
+  	$2.SetWith($1)
```
Comment (GuptaManan100) [inline go/vt/sqlparser/sql.y:810]: This doens't look a necessary change. It also messes up the indentation.

**Review by GuptaManan100 — CHANGES_REQUESTED** [review]

File: go/vt/sqlparser/ast.go [hunk 3]
```diff
@@ -2782,6 +2782,16 @@ type (
 		Geom     Expr
 	}
 
+	// PointPropertyType is an enum to get the types of geom format functions with possible values: XCordinate YCordinate Latitude Longitude
+
```
Comment (frouioui) [inline go/vt/sqlparser/ast.go:2786]: nit: we do not need a new line here

File: go/vt/sqlparser/constants.go [hunk 4]
```diff
@@ -453,6 +453,12 @@ const (
 	EnvelopeStr     = "st_envelope"
 	DimensionStr    = "st_dimension"
 	GeometryTypeStr = "st_geometrytype"
+
+	//PointPropertyType strings
```
Comment (frouioui) [inline go/vt/sqlparser/constants.go:457]:
```suggestion
	// PointPropertyType strings
```

**Review by frouioui — APPROVED** [review]

File: go/vt/sqlparser/ast.go (hunk 3 above)
Comment (Ayman161803) [reply go/vt/sqlparser/ast.go]: This new line has been removed. Few of my previous PRs introduced a similar newline after comments in ``ast.go``. I have removes these newlines as well.

**Pull request merged**